File: features/IoTCustomAutomationConfiguration/IoTCustomAutomationConfiguration.Business/Common/DataStructures/AutomationConfigurationData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace Cmf.Community.IoTCustomAutomationConfiguration.Actions.Common.DataStructures
{
    // <summary>
    // Support class that represents an extended custom property
    // </summary>
    [DataContract]
    public class AutomationConfigurationData
    {
        // <summary>
        // Automation Configuration Name
        // </summary>
        [DataMember]
        public string AutomationConfigurationName { get; set; }

        // <summary>
        // Automation Configuration Related Entity Name
        // </summary>
        [DataMember]
        public string AutomationConfigurationRelatedEntityName { get; set; }

        // <summary>
        // Automation Configuration Driver Friendly Name
        // </summary>
        [DataMember]
        public string AutomationConfigurationDriverFriendlyName { get; set; }

        // <summary>
        // Automation Configuration Values
        // </summary>
        [DataMember]
        public List<AutomationConfigurationValue> AutomationConfigurationValues { get; set; }

        // <summary>
        // Nested Automation Configuration Data
        // </summary>
        [DataMember]
        public List<AutomationConfigurationData> NestedAutomationConfigurationData { get; set; }

        // <summary>
        // Extended Information for Custom Behavior
        // </summary>
        [DataMember]
        public List<ExtendedCustomProperty> ExtendedCustomProperties { get; set; }
    }
}

File: features/IoTCustomAutomationConfiguration/IoTCustomAutomationConfiguration.IoT/src/connect-iot-custom-automationconfiguration-tasks/src/persistence/model/automationConfigurationData.ts
import { AutomationConfigurationValue } from "./automationConfigurationValue";
import { ExtendedCustomProperty } from "./extendedCustomProperty";


export interface AutomationConfigurationData {

    // Automation Configuration Name
    AutomationConfigurationName : string,
    // Automation Configuration Related Entity Name
    AutomationConfigurationRelatedEntityName: string,
    // Automation Configuration Driver Friendly Name
    AutomationConfigurationDriverFriendlyName: string,
    AutomationConfigurationValues: AutomationConfigurationValue[],
    // Nested Automation Configuration Data
    NestedAutomationConfigurationData: AutomationConfigurationData,
    /** Extended information for custom behavior */
    ExtendedCustomProperties: ExtendedCustomProperty[]
    /** Last Time the Material was updated */
    LastUpdate: string,

}

export enum MaterialStateEnum {
    Setup = "Setup",
    InProcess = "InProcess",
    Complete = "Complete",
    Aborting = "Aborting",
    Aborted = "Aborted"
}
